### ISSUE TITLE
Command output security concern

### ISSUE THREAD

**Issue by awiddersheim** [mickem/nscp#142]
I noticed that the command's output will displayed in certain circumstances. While I understand the reasoning, my concern is sometimes these commands will have passwords contained within them that you may not want users to see. Obviously, passing parameters itself is considered unsafe but wondering what the general consensus on this is. Here are some of the spots where the command line is output:

https://github.com/mickem/nscp/blob/master/modules/CheckExternalScripts/CheckExternalScripts.cpp#L484
https://github.com/mickem/nscp/blob/master/modules/CheckExternalScripts/CheckExternalScripts.cpp#L540
https://github.com/mickem/nscp/blob/master/modules/CheckExternalScripts/CheckExternalScripts.cpp#L546

**Comment by mickem** [comment]
Good point... 
I will see if I can remove some of the arguments from some commands.

One option long term might be to add some form of credentials store which could be used in the arguments?

**Comment by awiddersheim** [comment]
I can't say that I personally would find the credential store particularly useful. Mainly because our current infrastructure doesn't lend itself to easily maintaining such a thing on many disparate boxes. Others might though.

In all reality, for me, just having the messages like `Invalid response from command` and leaving out the actual command part would be more than enough for me to at least start to track down the problem. From there if I needed to see the actual command I could run the client in test mode for a few minutes to get more debug info in safer manner.

However, if removing the arguments from commands isn't difficult for you to do than I'd say go for that solution.

**Comment by mickem** [comment]
Removing ags from the return message would be easy, what would be more difficult would be removing it from logs as well... or well, that would be simple as well but difficult for people to see what amiss...

But if it is only returned message I guess it would pretty simple to do...

**Comment by awiddersheim** [comment]
Hrm, forgot about logs. Probably wouldn't want that stuff showing up there as well.

**Comment by mickem** [comment]
And that where I think we need a credential store otherweise it is impossible to know what a password is.
But when I say credential store I mean something along the lines of:


```
[/settings/passwords]
master password=<secret key>
admin_password=<OBFUSCATED HASH using <secret key>>

[/settings/external scripts/scripts]
check=my_ad_check --user admin --password ${admin_password}
```

But the quick fix which I will implemented for 0.4.3 is to remove the actual command line from the return message.

**Comment by mickem** [comment]
Resolved in next build, the only log entries which contain command lines are debug related so I think we can close this issue in regards to logging as well.
